File: app/src/main/java/com/example/testhexagongame/Color/ColorGenerator.java
package com.example.testhexagongame.Color;

import java.util.ArrayList;

import kotlin.jvm.functions.Function1;
import kotlin.jvm.functions.Function2;

public class ColorGenerator<T> {
    private final ArrayList<T> colors;
    private final Function2<Integer, Integer, Integer> randomNum;

    public ColorGenerator(ArrayList<T> colors, Function2<Integer, Integer, Integer> randomNum) {
        this.colors = colors;
        this.randomNum = randomNum;
    }

    public T getColorRandom() {
        return colors.get(randomNum.invoke(0, colors.size()-1));
    }
}

File: app/src/main/java/com/example/testhexagongame/Color/ColorGenerator.kt
package com.example.testhexagongame.Color

class ColorGenerator(
    private val listColor: List<String>,
    private val getRandom: (Int, Int) -> Int
    ) {
    fun getColorRandom(): String {
        return listColor[getRandom(0, listColor.size-1)]
    }
}